c7n/resources/s3.py

```diff
@@ -1372,6 +1372,144 @@ def process_bucket(self, bucket):
                 return {'Name': bucket['Name'], 'State': 'ReplicationConfigUpdated'}
 
 
+@filters.register('check-public-block')
+class FilterPublicBlock(Filter):
+    """Filter for s3 bucket public blocks
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: CheckForPublicAclBlock-Off
+                resource: s3
+                region: us-east-1
+                filters:
+                  - type: check-public-block
+                    scope: Any
+                    state: absent
+    """
+
+    schema = type_schema(
+        'check-public-block',
+        scope={
+            'type': 'string',
+            'enum': ['BlockPublicAcls', 'IgnorePublicAcls',
+                'BlockPublicPolicy', 'RestrictPublicBuckets', 'All', 'Any']},
+        state={
+            'type': 'string',
+            'enum': ['absent', 'present']},
```

Comment:
absent and present here feel wrong ... as the config as a whole is absent or present (where absent really means all settings are disabled).  It seems to make more sense (to me anyway) if these were `enable` and `disable`.

Reply:
this is on the filter and enable/disable are more action directives

Reply:
I agree that enable/disable are more action directives and absent/present are more filter directives. Happy to update though if there is a push for consistency.

Reply:
I guess in my head `absent` and `present` represent something different.  those simply declare the existence of something, not what the value is set to ... something can be present, but false.  How would folks feel about changing this to a boolean instead?  In this case, the absence of a configuration is equivalent to all settings set to false.

Reply:
I can see what you are saying here, especially with the way the account level filter is done where you would apply a boolean for each block.

Reply:
A change like that would alter the novelty of what is there now though, for example:

```
                  - type: check-public-block
                    scope: Any
                    state: absent
```
would probably become

```
          -or:
                  - type: check-public-block
                     BlockPublicPolicy: false
                  - type: check-public-block
                     RestrictPublicBuckets: false
                  - type: check-public-block
                     BlockPublicAcls: false
                  - type: check-public-block
                     IgnorePublicAcls: false
```
or are you thinking something more like?

```
                  - type: check-public-block
                    scope: Any
                    enabled: False
```

Reply:
yeah. i was just thinking the latter:
```
      - type: check-public-block
        scope: [Any]
        enabled: false
```
I like the brevity of the `scope` setting letting us filter on `Any` or `All` settings at once...

Reply:
and actually, seeing that ... maybe `scope` should allow either a single string, or an array of enums?  so it could be: `scope: Any` or `scope: BlockPublicAcls` or
```
    scope:
      - BlockPublicPolicy
      - BlockPublicAcls
```
having to specify `scope: [All]` or `scope: [Any]` seems odd.

Reply:
good call, I could see where enabled: false/true would make things clearer. I'll also look into how we can allow the single string or array of enums.

Reply:
Addressed in c99b47a, still not sure about the single string or array of enums option, from what I have seen it is either or. Will investigate some more though and see what can be done.